### ISSUE TITLE
Naming a Julia package

### ISSUE THREAD

**Issue by sbohora** [JuliaLang/METADATA.jl#8485]
I am trying to name a Julia package that I am currently developing. This is about a method on Semi-parametric Area Under the Curve (sAUC) model adjusting for discrete covariates.  I initially wanted to call it `sAUC.jl` or `SemiparametricAUC.jl`. I am concerned if these names would be in line the naming convention of Julia package. I do not see an example of package name with abbreviated upper cases. Any idea if these names are accepted as package name?
Thanks.
I really appreciate it.

@ChrisRackauckas

**Comment by quinnj** [comment]
`SemiparametricAUC.jl` sounds fine to me. `sAUC.jl` is also probably ok, but typically packages don't lowercase the first letter. Either way is probably fine since they're unlikely to clash w/ other names. Another strategy is to choose a more general name, `AreaUnderCurves.jl`, or something, and then be explicit in the documentation about the methods or algorithms implemented.

**Comment by ChrisRackauckas** [comment]
>sAUC.jl is also probably ok

Usually abbreviations are not accepted. I am trying to find a reference for that but I can't find it right now.

**Comment by quinnj** [comment]
I'd say it depends: we have `CSV.jl` for example, mainly because `CSV` is a very wide-spread, well-known acronym for comma-separate values. If an acronym is unambiguous enough, I think it's fine. For `sAUC.jl`, it seems like it's unlikely to collide w/ another acronym, I'm not sure if `SAUC.jl` would mean anything or not, but that would be a potential concern. As-is though, I'd worry more about if someone who was looking for semi-parametric area-under-the-curve models would recognize `sAUC.jl` or not.

**Comment by dpsanders** [comment]
Google does not seem to know what sauc is, so that does not seem like a
good option in my opinion.

**Comment by ChrisRackauckas** [comment]
>we have CSV.jl for example, mainly because CSV is a very wide-spread, well-known acronym for comma-separate values

Yes, but CSV is well-known enough that most people will know generally what the package is for just by its name. I believe that was a criteria mentioned by @tkelman and it's probably a good idea. `SAUC.jl`, if it's a well-known acronym, it's domain-specific, and the names shouldn't have that kind of requirement. 

Generally it's all different interpretations of this:

http://docs.julialang.org/en/release-0.5/manual/packages/#guidelines-for-naming-a-package

I think these come up here:

>Avoid jargon. In particular, avoid acronyms unless there is minimal possibility of confusion.

>Err on the side of clarity, even if clarity seems long-winded to you.
